routes/users.js

```diff
@@ -13,42 +12,27 @@ router.use(express.json());
 
 // POST /users/login
 router.post("/users/login", async (req, res) => {
-  const token = req.headers['token'];
-  if (!token) {
-    return respondWithError(res, 401, "No token provided");
+  const firebaseToken = req.headers['firebase_token'] || null;
+  const discordCode = req.headers['discord_code'] || null;
+  if (!firebaseToken && !discordCode) {
+    return respondWithError(res, 401, 'No token provided or invalid token');
   }
-  try {
-    const decodedToken = await admin.auth().verifyIdToken(token);
-    const email = decodedToken.email;
-    const firebaseId = decodedToken.uid;
-
-    if (!decodedToken.email_verified) {
-      return respondWithError(res, 403, "Email not verified");
-    }
 
-    const application = await applService.getApplicationByEmail(email);  
-    if (!application || !application.status === 'approved') {
-      return respondWithError(res, 403, "Application is not approved or does not exist");
-    }
+  try {
+    let result;
 
-    let user = await userService.getUserByEmail(email);
-    if (!user) {
-      user = await userService.createNewMemberUser(
-        application.name,
-        email,
-        application.about,
-        application.languages, 
-        application.discord_nickname
-      );
+    if (firebaseToken) {
+      result = await authService.handleFirebaseAuth(firebaseToken);
+    } else if (discordCode) {
+      result = await authService.handleDiscordAuth(discordCode);
     }
 
-    await userService.updateUserById(user.id, {firebase_id: firebaseId});  
+    res.cookie('userUID', result.user.firebaseId, { httpOnly: true, secure: true, sameSite: 'none' });
+    res.status(200).json({ user: result.user });
 
-    res.cookie('userUID', firebaseId, { httpOnly: true, secure: true, sameSite: 'none' });
-    res.status(200).json({ user: user });
   } catch (error) {
-    console.error(error);
-    return respondWithError(res, 401, "Unauthorized");
+    console.error("Authentication Error:", error.message);
+    return respondWithError(res, error.status || 500, error.message);
```

Comment:
very nice solution with the error object, but what happens if the error thrown was not the one you were expected? are we sure we want to return ANY error message to the frontend?
I think it would be safer process custom errors separately and fall back to default 500 in other cases